src/client/clientmap.cpp

```diff
@@ -767,15 +767,12 @@ void ClientMap::renderMap(video::IVideoDriver* driver, s32 pass)
 		if (is_frustum_culled(mesh_sphere_center, mesh_sphere_radius))
 			continue;
 
-		v3f block_pos_r = intToFloat(block->getPosRelative() + MAP_BLOCKSIZE / 2, BS);
-
-		float d = camera_position.getDistanceFrom(block_pos_r);
-		d = MYMAX(0,d - BLOCK_MAX_RADIUS);
-
 		// Mesh animation
 		if (pass == scene::ESNRP_SOLID) {
-			// Pretty random but this should work somewhat nicely
-			bool faraway = d >= BS * 50;
+			// 50 nodes is pretty arbitrary but it should work somewhat nicely
```

Comment:
I wonder if this is still a good number today. We've since increased the viewing_range to 190, and generally Minetest runs much better with larger viewing ranges.

Perhaps it's better to set this to 0.5 * viewing_range or something.... Anyway, for another PR.

Reply:
Yes. Someone should start playing around with many different animated nodes in many mesh chunks, and look how the value should be changed and if render performance is struggling.
Note that if we're faraway, the block is still every 5 to 100 frames (picked randomly, see `MapBlockMesh::animate`).

Anyway, for another PR.